Correcting some grammatical errors of  README.md (#1911)
iluwatar/java-design-patterns: README.md

# Introduction

> Design patterns are the best-formalized practices a programmer can use to

Reviewer: I think the intention here is to say `Design patterns are the best, formalized practices...`

Author: Ok correcting.

 - Using tags such as `Performance`, `Gang of Four` or `Data access`.
 - Using pattern categories, `Creational`, `Behavioral`, and others.

Hopefully, you find the object-oriented solutions presented on this site useful

> in your architectures and have as much fun learning them as we had when we were developing them.

Reviewer: ```suggestion
in your architectures and have as much fun learning them as we had while developing them.
```
